### PR TITLE
KOGITO-4771 - Update SW to v0.6 and its java-sdk to 2.0.0.final

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tsurdilo@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by tsurdilo** [comment]
@ricardozanini can you let me know which Jira should be used for this? It should also include same update for kogito examples which I'm working on next

File: kogito-build-parent/pom.xml [hunk 1]
```diff
@@ -148,7 +148,7 @@
     <version.io.micrometer>1.6.1</version.io.micrometer>
     <version.io.rest-assured>4.3.0</version.io.rest-assured>
     <version.io.rest-assured.springboot>3.3.0</version.io.rest-assured.springboot>    <!-- Manually override to 3.3.0 because Spring boot uses rest-assured-common@3.3.0, otherwise error: The type io.restassured.common.mapper.TypeRef cannot be resolved. It is indirectly referenced from required .class files -->
-    <version.io.serverlessworkflow>1.0.3.Final</version.io.serverlessworkflow>
+    <version.io.serverlessworkflow>2.0.0.Final</version.io.serverlessworkflow>
```
Comment (ricardozanini) [inline kogito-build-parent/pom.xml:151]: There's an exclusion in this file removing the `commons-logging` dependency from the SDK. Could you please also remove it?

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
THANK YOU MAN!
Opened [KOGITO-4771](https://issues.redhat.com/browse/KOGITO-4771) to track this one.

File: kogito-build-parent/pom.xml (hunk 1 above)
Comment (tsurdilo) [reply kogito-build-parent/pom.xml:151]: removed

**Review by ricardozanini — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]

**Comment by ricardozanini** [comment]
Jenkins run native

**Comment by ricardozanini** [comment]
@danielezonca error in OptaPlanner. Our build went ok. Ok to merge?

**Comment by danielezonca** [comment]
Failure is unrelated and after -runtime so merging

**Pull request merged**